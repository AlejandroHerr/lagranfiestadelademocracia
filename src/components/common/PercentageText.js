import React from 'react';
import PropTypes from 'prop-types';

import Percentage from './Percentage';
import NormalText from './NormalText';

const PercentageText = ({ children, ...props }) => (
  <Percentage renderComponent={NormalText} {...props}>
    {children}
  </Percentage>
);

PercentageText.propTypes = {
  children: PropTypes.number.isRequired,
};

export default PercentageText;
