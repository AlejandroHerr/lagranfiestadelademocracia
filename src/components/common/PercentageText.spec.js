import React from 'react';
import renderer from 'react-test-renderer';

import PercentageText from './PercentageText';

describe('components/component/PercentageText', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PercentageText renderComponent={props => <div {...props} />}>{1}</PercentageText>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
