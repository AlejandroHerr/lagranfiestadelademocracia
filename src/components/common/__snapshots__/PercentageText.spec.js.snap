// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/component/PercentageText renders correctly 1`] = `
<div>
  100%
</div>
`;
